packages/middleware-retry/src/index.spec.ts

```diff
@@ -26,28 +26,28 @@ describe("retryMiddleware", () => {
   });
 
   it("should stop retrying when the the maximum number of retries is reached", async () => {
-    const maxRetries = 3;
+    const maxAttempts = 3;
     const error = new Error();
     error.name = "ProvisionedThroughputExceededException";
     const next = jest.fn().mockRejectedValue(error);
-    const retryHandler = retryMiddleware(resolveRetryConfig({ maxRetries }))(
+    const retryHandler = retryMiddleware(resolveRetryConfig({ maxAttempts }))(
       next
     );
 
     await expect(
       retryHandler({ input: {}, request: new HttpRequest({}) })
     ).rejects.toMatchObject(error);
 
-    expect(next.mock.calls.length).toBe(maxRetries + 1);
+    expect(next.mock.calls.length).toBe(maxAttempts + 1);
```

Comment:
`maxAttempts` has different meaning to `maxRetries` from what I see. When `maxRetries` is set to 3, the total attempt is 4 times. Whereas for `maxAttempts`, you don't expect the total attempt to be maxAttempts + 1.

Reply:
Fixed in [commit](https://github.com/aws/aws-sdk-js-v3/pull/1244/commits/7d63dea111ba0e0f880ad309406f3165a613b43d)